src/hooks.spec.tsx

```diff
@@ -0,0 +1,174 @@
+import { ReactSDKClient } from './client';
+import { useFeature } from './hooks';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+const MyComponent = () => {
+  const [ isEnabled ] = useFeature('feature1');
+  return <>{`${isEnabled ? 'true' : 'false'}`}</>;
+}
+
+const MyComponentTimeout100 = () => {
+  const [ isEnabled, _, clientReady ] = useFeature('feature1', { timeout: 100 });
+  return <>{`${isEnabled ? 'true' : 'false'}|${clientReady}`}</>;
+}
+
+const MyComponentAutoUpdate = () => {
+  const [ isEnabled ] = useFeature('feature1', { autoUpdate: true });
+  return <>{`${isEnabled ? 'true' : 'false'}`}</>;
+}
+
+describe('hooks', () => {
+  const featureVariables = {
+    foo: 'bar',
+  };
+  let isFeatureEnabledMock: jest.Mock;
+  let resolver: any;
+  let optimizelyMock: ReactSDKClient;
+  let userUpdateCallbacks: Array<() => void>;
+  let notificationListenerCallbacks: Array<() => void>;
+  
+  beforeEach(() => {
+    const onReadyPromise = new Promise((resolve, reject) => {
+      resolver = {
+        reject,
+        resolve,
+      }
+    });
+
+    isFeatureEnabledMock = jest.fn();
+    userUpdateCallbacks = [];
+    notificationListenerCallbacks = [];
+
+    optimizelyMock = ({
+      onReady: jest.fn().mockImplementation(config => onReadyPromise),
+      getFeatureVariables: jest.fn().mockImplementation(() => featureVariables),
+      isFeatureEnabled: isFeatureEnabledMock,
+      onUserUpdate: jest.fn().mockImplementation(handler => {
+        userUpdateCallbacks.push(handler);
+        return () => {};
+      }),
+      notificationCenter: {
+        addNotificationListener: jest.fn().mockImplementation((type, handler) => {
+          notificationListenerCallbacks.push(handler);
+        }),
+        removeNotificationListener: jest.fn().mockImplementation(id => {}),
+      },
+      user: {
+        id: 'testuser',
+        attributes: {},
+      },
+    } as unknown) as ReactSDKClient
+  })
+
+  describe('useFeature', () => {
+    it('should render true when the feature is enabled', async () => {
+      isFeatureEnabledMock.mockReturnValue(true);
+      const component = Enzyme.mount(
+        <OptimizelyProvider optimizely={optimizelyMock}>
+          <MyComponent />
+        </OptimizelyProvider>,
+      )
+      resolver.resolve({ success: true });
+      await optimizelyMock.onReady();
+      component.update();
+      expect(component.text()).toBe('true')
+    });
+
+    it('should render false when the feature is disabled', async () => {
+      isFeatureEnabledMock.mockReturnValue(false);
+      const component = Enzyme.mount(
+        <OptimizelyProvider optimizely={optimizelyMock}>
+          <MyComponent />
+        </OptimizelyProvider>,
+      );
+      resolver.resolve({ success: true });
+      await optimizelyMock.onReady();
+      component.update();
+      expect(component.text()).toBe('false');
+    });
+
+    it.skip('should respect the timeout option passed', async () => {
```

Comment:
Is this not working yet? Should we have an explicit test for that option passed as `false` or is implicit undefined via the test above sufficient?

Reply:
Going to update this now! Thanks for asking.

It's a little tricky with a mock client, but I think I know what I wanna do.